invoices/sql_store.go

```diff
@@ -925,8 +925,10 @@ func (i *SQLStore) QueryInvoices(ctx context.Context,
 			}
 
 			if q.CreationDateEnd != 0 {
+				// We need to add 1 to the end date as we're
+				// checking less than the end date in SQL.
 				params.CreatedBefore = sqldb.SQLTime(
-					time.Unix(q.CreationDateEnd, 0).UTC(),
+					time.Unix(q.CreationDateEnd+1, 0).UTC(),
```

Comment:
Why do we need the `+1` here if we were checking `created_at <=` below?

Perhaps a simple test case can better illustrate the issue can the fix.

Reply:
This is already covered (and caught) in the following itest: ```make itest backend=bitcoind dbbackend=postgres nativesql=true icase=list_payments```

Reply:
The +1 is needed because we round down with `time.Unix` but we store exact time stamps in the database.